File: go/Zadanie1/stack.go
// stack.go
package main

import "fmt"

type Stack struct {
	arr      []string
	top      int
	capacity int
	silent   bool
}

func NewStack(capacity int, silent bool) *Stack {
	return &Stack{
		arr:      make([]string, capacity),
		capacity: capacity,
		top:      -1,
		silent:   silent,
	}
}

func (s *Stack) Find(value string) int {
	for i := 0; i <= s.top; i++ {
		if s.arr[i] == value {
			return i
		}
	}
	return -1
}

func (s *Stack) Print() {
	for i := 0; i <= s.top; i++ {
		fmt.Print(s.arr[i])
		if i < s.top {
			fmt.Print(", ")
		}
	}
	fmt.Println()
}

func (s *Stack) Push(value string) {
	if s.top >= s.capacity-1 {
		if !s.silent {
			fmt.Println("Стек переполнен!")
		}
		return
	}
	s.top++
	s.arr[s.top] = value
	if !s.silent {
		fmt.Printf("Элемент %s добавлен в стек\n", value)
	}
}

func (s *Stack) Pop() {
	if s.top < 0 {
		if !s.silent {
			fmt.Println("Стек пуст!")
		}
		return
	}
	if !s.silent {
		fmt.Printf("Элемент %s удалён из стека\n", s.arr[s.top])
	}
	s.top--
}

func (s *Stack) Top() string {
	if s.top < 0 {
		if !s.silent {
			fmt.Println("Стек пуст!")
		}
		return ""
	}
	return s.arr[s.top]
}

func (s *Stack) IsEmpty() bool {
	return s.top < 0
}

func (s *Stack) TopIndex() int {
	return s.top
}

func (s *Stack) GetAt(index int) string {
	if index < 0 || index > s.top {
		return ""
	}
	return s.arr[index]
}

func (s *Stack) Size() int {
	return s.top + 1
}

File: с++/Zadanie1/Zadanie1/stack.cpp
#include <iostream>
#include "stack.h"
using namespace std;

// Конструктор
Stack::Stack(int size, bool silentMode) {
    capacity = size;
    arr = new string[capacity];
    topIndex = -1;
    silent = silentMode;

    // Инициализируем все элементы пустыми строками
    for (int i = 0; i < capacity; i++) {
        arr[i] = "";
    }
}

// Деструктор
Stack::~Stack() {
    if (arr != nullptr) {
        delete[] arr;
        arr = nullptr;
    }
}

// Поиск элемента в стеке по значению
int findInStack(const Stack& stack, const std::string& value) {
    for (int i = 0; i <= stack.topIndex; i++) {
        if (stack.arr[i] == value) {
            return i;
        }
    }
    return -1;
}

// Вывод всех элементов стека
void printStack(const Stack& stack) {
    for (int i = 0; i <= stack.topIndex; i++) {
        cout << stack.arr[i];
        if (i < stack.topIndex) {
            cout << ", ";
        }
    }
    cout << endl;
}

// Добавление элемента в стек
void Stack::push(const string& value) {
    if (topIndex >= capacity - 1) {
        if (!silent) {
            cout << "Стек переполнен!" << endl;
        }
        return;
    }
    topIndex++;
    arr[topIndex] = value;
    if (!silent) {
        cout << "Элемент " << value << " добавлен в стек" << endl;
    }
}

// Удаление элемента из стека
void Stack::pop() {
    if (topIndex < 0) {
        if (!silent) {
            cout << "Стек пуст!" << endl;
        }
        return;
    }
    if (!silent) {
        cout << "Элемент " << arr[topIndex] << " удалён из стека" << endl;
    }
    topIndex--;
}

// Просмотр верхнего элемента
string Stack::top() {
    if (topIndex < 0) {
        if (!silent) {
            cout << "Стек пуст!" << endl;
        }
        return "";
    }
    return arr[topIndex];
}

// Проверка, пуст ли стек
bool Stack::isEmpty() {
    return topIndex < 0;
}

// Конструктор копирования
Stack::Stack(const Stack& other) {
    capacity = other.capacity;
    topIndex = other.topIndex;
    silent = other.silent;
    arr = new string[capacity];
    for (int i = 0; i <= topIndex; i++) {
        arr[i] = other.arr[i];
    }
}

// Оператор присваивания
Stack& Stack::operator=(const Stack& other) {
    if (this != &other) {
        delete[] arr;
        capacity = other.capacity;
        topIndex = other.topIndex;
        silent = other.silent;
        arr = new string[capacity];
        for (int i = 0; i <= topIndex; i++) {
            arr[i] = other.arr[i];
        }
    }
    return *this;
}